### PR TITLE
Add nix derivation

### PR DESCRIPTION
**Pull request by joris-burgers**
Adds a specialized nix derivation for vaultenv.

Vaultenv can now be included in other projects by building `./nix/release.nix`. The nix dependencies are now also managed using [`niv`](https://github.com/nmattia/niv).

This PR also bumps the GHC version to 8.8.4.

### CODE REVIEW

File: nix/release.nix [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# This file wires up the different vaultenv releases that are available.
```
Comment (FPtje) [inline nix/release.nix:1]: There is only one release available, does this comment still apply?

File: nix/release.nix [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+rec {
+  pkgs = import ./nixpkgs-pinned.nix {};
```
Comment (FPtje) [inline nix/release.nix:3]: This doesn't seem like it should be member of the set of released packages. Perhaps it fits better in a let-binding.

File: stack.yaml.lock [hunk 3]
```diff
@@ -4,9 +4,4 @@
 #   https://docs.haskellstack.org/en/stable/lock_files
 
 packages: []
-snapshots:
-- completed:
-    size: 523700
-    url: https://raw.githubusercontent.com/commercialhaskell/stackage-snapshots/master/lts/14/7.yaml
-    sha256: 8e3f3c894be74d71fa4bf085e0a8baae7e4d7622d07ea31a52736b80f8b9bb1a
-  original: lts-14.7
+snapshots: []
```
Comment (FPtje) [inline stack.yaml.lock:7]: I think this file can be removed entirely when using nix 👍

File: vaultenv.nix [hunk 4]
```diff
@@ -0,0 +1,76 @@
+{ haskellPackages
+, mkDerivation
+, lib
+, pkgs
+, glibcLocales
```
Comment (FPtje) [inline vaultenv.nix:5]: Are you sure the `glibcLocales` are needed for this derivation?

**Review by FPtje — CHANGES_REQUESTED** [review]
Thanks for creating better nix packaging and upgrading ghc! Some comments.

File: nix/release.nix (hunk 1 above)
Comment (joris-burgers) [reply nix/release.nix]: Well spotted, no. This PR doesn't use building static binaries, so we only have a single release.

File: nix/release.nix (hunk 2 above)
Comment (joris-burgers) [reply nix/release.nix:2]: I moved it to a let binding.

File: stack.yaml.lock (hunk 3 above)
Comment (joris-burgers) [reply stack.yaml.lock]: YES WE CAN!

File: vaultenv.nix (hunk 4 above)
Comment (joris-burgers) [reply vaultenv.nix:5]: Not sure, but I didn't test whether `vaultenv` still suffers from the `commitAndReleaseBuffer: invalid argument (invalid character)` error message without it, that's why I included this.

**Review by FPtje — APPROVED** [review]
LGTM! :+1:

**Pull request merged**